Twig/DefaultApplyingNodeVisitor.php

```diff
@@ -85,18 +85,18 @@ public function enterNode(Node $node, Environment $env): Node
                 // remove the replacements from the test node
                 $testNodeArguments    = iterator_to_array($testNode->getNode('arguments'));
                 $testNodeArguments[0] = new ArrayExpression([], $lineno);
-                $testNode->setNode('arguments', new Node($testNodeArguments));
+                $testNode->setNode('arguments', new Nodes($testNodeArguments));
 
                 // wrap the default node in a |replace filter
                 $defaultNode = new FilterExpression(
                     $arguments[0],
-                    new ConstantExpression('replace', $lineno),
-                    new Node([$wrappingNodeArguments[0]]),
+                    $env->getFilter('replace'),
```

Comment:
`Twig\Environment::getFilter` method is considered as `@internal`, however I didn't find better way to retrieve TwigFilter instance.